### ISSUE TITLE
Is there a properflow to close the popup after a successful SignIn using signinPopup()

### ISSUE THREAD

**Issue by ananta** [authts/react-oidc-context#382]
I haven't configured much and went with a basic example.

I am pretty new to OIDC authentication and authorization, but this is what I am getting. 

https://user-images.githubusercontent.com/12180395/169503031-f48ce9a2-71c0-43e5-8567-c6a4f54701c9.mov

**Comment by pamapa** [comment]
After a successful sign you can use [onSigninCallback](https://github.com/authts/react-oidc-context/blob/main/src/AuthProvider.tsx#L33) to do what ever needs to be done for your application

**Comment by ananta** [comment]
Thank you so much for your reply.
I also expected the same thing, but the callback didn't fire on mine.

It turns out that after successful login with the oidc-server, react-oidc-context was unable to switch to the `logged-in` state. As I wasn't getting any error to debug, I initiated this issue.

I created a custom auth flow with the bare [oidc-client-ts](https://github.com/authts/oidc-client-ts) library to debug this issue with my application. It turns out the oidc-client-ts couldn't access the state after a successful redirect from the oidc-server. 

Why? Because of the '#' in the URL. My application was using a HashRouter from [react-router-dom](https://github.com/remix-run/react-router). 

Because of this reason, the library was throwing a `No state in response` error. But I couldn't catch the error with this library. 

If anyone is having the same issue, there you go :)
